src/Analyzer/Pass/Statement/YodaCondition.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @author Alexey Kolpakov https://github.com/alhimik45 <alhimik45@gmail.com>
+ */
+
+namespace PHPSA\Analyzer\Pass\Statement;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\Equal;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\BinaryOp\NotEqual;
+use PhpParser\Node\Expr\BinaryOp\NotIdentical;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Scalar;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Stmt\Case_;
+use PhpParser\Node\Stmt\Do_;
+use PhpParser\Node\Stmt\ElseIf_;
+use PhpParser\Node\Stmt\For_;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\While_;
+use PHPSA\Analyzer\Pass;
+use PHPSA\Context;
+
+class YodaCondition implements Pass\AnalyzerPassInterface
+{
+    /**
+     * @param $stmt
+     * @param Context $context
+     * @return bool
+     */
+    public function pass($stmt, Context $context)
+    {
+        $condition = $stmt->cond;
+
+        if ($stmt instanceof For_ && count($stmt->cond) > 0) { // For is the only one that has an array as condition
```

Comment:
@K-Phoen I would think if anything we should show a message saying that you should make those conditions into one (like you have to do with all other statements that use conditions). It makes no sense to me to have "multiple conditions" if you can merge them with boolean and.